WordPress/Classes/Services/MediaService.m

```diff
@@ -57,7 +57,10 @@ - (Media *)createMediaWith:(id<ExportableAsset>)exportable
         }
         
         if (blog.allowedFileTypes != nil) {
-            allowedFileTypes = blog.allowedFileTypes;
+            // HEIC isn't supported when uploading an image, so we filter it out (http://git.io/JJAae)
+            NSMutableSet *mutableAllowedFileTypes = [blog.allowedFileTypes mutableCopy];
```

Comment:
@leandroalonso Since the constraint is for all types of sites, I wonder if we should consider placing this condition in [`MediaAssetExporter`](https://github.com/wordpress-mobile/WordPress-iOS/blob/issue/14680_fix_heic_upload/WordPress/Classes/Utility/Media/MediaAssetExporter.swift) instead?